lib/tapioca/gem/listeners/foreign_constants.rb

```diff
@@ -0,0 +1,38 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Tapioca
+  module Gem
+    module Listeners
+      class ForeignConstants < Base
+        extend T::Sig
+
+        include Runtime::Reflection
+
+        private
+
+        sig { override.params(event: ScopeNodeAdded).void }
+        def on_scope(event)
+          mixin = event.constant
+          return if Class === mixin # Classes can't be mixed into other constants
+
+          # There are cases where we want to process constants not declared by the current
```

Comment:
@paracycle Updated the comment here! It is almost identical to what you suggested with a couple of small wording changes.